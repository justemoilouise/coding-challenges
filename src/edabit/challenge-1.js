import * as React from 'react';
import * as styles from '../styles/challenge.module.css';
import { Pill } from '../components/pill/pill';
import { convertBinaryTextToNumber } from './challenge';

export const Challenge = () => {
  const inputRef = React.useRef();

  const [data, setData] = React.useState();

  const handleClick = () => {
    if (!inputRef?.current) {
      return null;
    }

    const input = inputRef.current.value.toLowerCase().split(' ').filter(x => ['zero', 'one'].includes(x));
    const output = convertBinaryTextToNumber(input);

    setData({
      input,
      output,
    });
    inputRef.current.value = '';
  };

  return (
    <div className={styles.container}>
      <div className={styles.flexContainer}>
        Input:&nbsp;<input ref={inputRef} className={styles.inputField} />
        <button className={styles.button} onClick={handleClick}>Convert</button>
      </div>
      {data && (
        <>
          <div className={styles.flexContainer}>
            Given:&nbsp;
            {data.input.map((num, i) => (<Pill key={i} variant='secondary'>{num}</Pill>))}
          </div>
          <div className={styles.flexContainer}>
            Output:&nbsp;<strong>{data.output}</strong>
          </div>
        </>
      )}
    </div>
  );
};
