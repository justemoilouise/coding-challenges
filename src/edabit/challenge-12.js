import * as React from 'react';
import * as styles from '../styles/challenge.module.css';
import classNames from 'classnames';
import { Pill } from '../components/pill/pill';
import { getCombinations } from './challenge';

export const Challenge = () => {
  const [input, setInput] = React.useState([]);
  const [output, setOutput] = React.useState();

  const onKeyDown = evt => {
    if (evt.code === 'Enter') {
      const currentTarget = evt.currentTarget;
      setInput(input.concat(parseInt(currentTarget.value)));
      currentTarget.value = '';
    }
  };

  const onClick = () => {
    setOutput(getCombinations(input));
  };

  return (
    <div className={classNames(styles.container, styles.gridContainer)}>
      <div>
        <div className={styles.flexContainer}>
          Input:&nbsp;
          <input className={styles.inputField} onKeyDown={onKeyDown} />
          <button className={styles.button} onClick={onClick}>
            Submit
          </button>
        </div>
        {input.length > 0 && (
          <div className={styles.flexContainer}>
            {input.map((value, index) => (
              <Pill key={index} variant='secondary'>
                {value}
              </Pill>
            ))}
          </div>
        )}
      </div>
      <div className={styles.flexContainer}>
        Output:&nbsp;<strong>{output}</strong>
      </div>
    </div>
  );
};
