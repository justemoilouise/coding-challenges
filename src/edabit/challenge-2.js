import * as React from 'react';
import * as styles from '../styles/challenge.module.css';
import classNames from 'classnames';
import { Pill } from '../components/pill/pill';
import { getFibonacci } from './challenge';

export const Challenge = () => {
  const inputRef = React.useRef();

  const [seq, setSeq] = React.useState();

  const handleClick = () => {
    if (!inputRef?.current) {
      return null;
    }

    setSeq(undefined);

    const input = parseInt(inputRef.current.value);
    if (input) {
      setSeq(getFibonacci(input));
    }
  };

  return (
    <div className={classNames(styles.container, styles.gridContainer)}>
      <div className={styles.flexContainer}>
        Input:&nbsp;<input type='number' min={0} ref={inputRef} className={styles.inputField} />
        <button className={styles.button} onClick={handleClick}>Submit</button>
      </div>
      {seq && (
        <div>
          Fibonacci Sequence:&nbsp;
          {seq.map((num, i) => (<Pill key={i} variant='secondary'>{num}</Pill>))}
        </div>
      )}
    </div>
  );
};
