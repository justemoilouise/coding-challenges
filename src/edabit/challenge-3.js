import * as React from 'react';
import * as styles from '../styles/challenge.module.css';
import classNames from 'classnames';
import { Pill } from '../components/pill/pill';
import { boomSeven } from './challenge';

export const Challenge = () => {
  const [input, setInput] = React.useState([]);
  const [output, setOutput] = React.useState([]);

  const onKeyDown = evt => {
    if (evt.code === 'Enter') {
      const currentTarget = evt.currentTarget;
      setInput(input.concat(parseInt(currentTarget.value)));
      currentTarget.value = '';
    }
  };

  const onClick = () => {
    setOutput(boomSeven(input));
  };

  return (
    <div className={classNames(styles.container, styles.gridContainer)}>
      <div>
        <div className={styles.flexContainer}>
          Input:&nbsp;
          <input type='number' className={styles.inputField} onKeyDown={onKeyDown} />
          <button className={styles.button} onClick={onClick}>
            Submit
          </button>
        </div>
        {input.length > 0 && (
          <div className={styles.flexContainer}>
            {input.map((value, index) => (
              <Pill key={index} variant='secondary'>
                {value}
              </Pill>
            ))}
          </div>
        )}
      </div>
      {output && (
        <div>
          <div className={styles.flexContainer}>Output:</div>
          {output.length > 0 && (
            <div className={styles.flexContainer}>
              <strong>{output}</strong>
            </div>
          )}
        </div>
      )}
    </div>
  );
};
