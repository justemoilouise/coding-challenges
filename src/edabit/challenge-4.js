import * as React from 'react';
import * as styles from '../styles/challenge.module.css';
import classNames from 'classnames';
import { towerOfHanoi } from './challenge';

export const Challenge = () => {
  const inputRef = React.useRef();

  const [output, setOutput] = React.useState([]);

  const onClick = () => {
    const value = inputRef.current.value;
    if (value > 0) {
      setOutput(towerOfHanoi(inputRef.current.value));
    }
  };

  return (
    <div className={classNames(styles.container, styles.gridContainer)}>
      <div className={styles.flexContainer}>
        Input:&nbsp;
        <input type='number' min={1} ref={inputRef} className={styles.inputField} />
        <button className={styles.button} onClick={onClick}>
          Play
        </button>
      </div>
      {output && output.length > 0 && (
        <div>
          <div className={styles.flexContainer}>Output:</div>
          <ul>
            {output.map((o, i) => (
              <li key={i}>{o}</li>
            ))}
          </ul>
        </div>
      )}
    </div>
  );
};
