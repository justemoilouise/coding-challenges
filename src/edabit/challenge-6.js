import * as React from 'react';
import * as styles from '../styles/challenge.module.css';
import classNames from 'classnames';
import { getNumberPoliteness } from './challenge';

export const Challenge = () => {
  const inputRef = React.useRef();

  const [output, setOutput] = React.useState([]);

  const onClick = () => {
    const value = inputRef.current.value;
    if (value > 0) {
      const politeness = getNumberPoliteness(inputRef.current.value);
      setOutput(politeness.length > 0);
    }
  };

  return (
    <div className={classNames(styles.container, styles.gridContainer)}>
      <div className={styles.flexContainer}>
        Input:&nbsp;
        <input type='number' min={1} ref={inputRef} className={styles.inputField} />
        <button className={styles.button} onClick={onClick}>
          Submit
        </button>
      </div>
      <div>Output:&nbsp;{output ? 'Yes' : 'No'}</div>
    </div>
  );
};
