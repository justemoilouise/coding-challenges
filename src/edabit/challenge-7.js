import * as React from 'react';
import * as styles from '../styles/challenge.module.css';
import classNames from 'classnames';
import { oddishOrEvenish } from './challenge';

export const Challenge = () => {
  const inputRef = React.useRef();

  const [output, setOutput] = React.useState();

  const onClick = () => {
    const value = inputRef.current.value;
    if (value > 0) {
      setOutput(oddishOrEvenish(value));
    }
  };

  return (
    <div className={classNames(styles.container, styles.gridContainer)}>
      <div className={styles.flexContainer}>
        Input:&nbsp;<input type='number' min={1} ref={inputRef} className={styles.inputField} />
        <button className={styles.button} onClick={onClick}>Submit</button>
      </div>
      <div>
        Output:&nbsp;<strong>{output}</strong>
      </div>
    </div>
  );
};
