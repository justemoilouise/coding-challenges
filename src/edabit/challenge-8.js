import * as React from 'react';
import * as styles from '../styles/challenge.module.css';
import classNames from 'classnames';
import { getAvgSpeed } from './challenge';

export const Challenge = () => {
  const upTimeRef = React.useRef();
  const upRateRef = React.useRef();
  const downRateRef = React.useRef();
  const [output, setOutput] = React.useState();

  const onClick = () => {
    if (!upRateRef.current || !downRateRef.current) {
      return;
    }

    setOutput(getAvgSpeed(
      parseInt(upRateRef.current.value),
      parseInt(downRateRef.current.value)
    ));
  };

  return (
    <div className={classNames(styles.container, styles.gridContainer)}>
      <div>
        <div className={styles.flexContainer}>
          Uphill Time:&nbsp;<input ref={upTimeRef} type='number' className={styles.inputField} />
        </div>
        <div className={styles.flexContainer}>
          Uphill Rate:&nbsp;<input ref={upRateRef} type='number' className={styles.inputField} />
        </div>
        <div className={styles.flexContainer}>
          Downhill Rate:&nbsp;<input ref={downRateRef} type='number' className={styles.inputField} />
        </div>
        <button className={styles.button} onClick={onClick}>Submit</button>
      </div>
      {output && (
        <div className={styles.flexContainer}>
          Output: <strong>{output}</strong>
        </div>
      )}
    </div>
  );
};
