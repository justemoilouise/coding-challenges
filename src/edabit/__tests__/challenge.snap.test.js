import React from 'react';
import renderer from 'react-test-renderer';
import { Challenge as Challenge1 } from '../challenge-1';
import { Challenge as Challenge2 } from '../challenge-2';
import { Challenge as Challenge3 } from '../challenge-3';
import { Challenge as Challenge4 } from '../challenge-4';
import { Challenge as Challenge5 } from '../challenge-5';
import { Challenge as Challenge6 } from '../challenge-6';
import { Challenge as Challenge7 } from '../challenge-7';
import { Challenge as Challenge8 } from '../challenge-8';
import { Challenge as Challenge9 } from '../challenge-9';
import { Challenge as Challenge10 } from '../challenge-10';
import { Challenge as Challenge11 } from '../challenge-11';
import { Challenge as Challenge12 } from '../challenge-12';
import { Challenge as Challenge13 } from '../challenge-13';

describe('Edabit challenges snapshots', () => {
  it('Challenge 1', () => {
    const tree = renderer
      .create(<Challenge1 />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Challenge 2', () => {
    const tree = renderer
      .create(<Challenge2 />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Challenge 3', () => {
    const tree = renderer
      .create(<Challenge3 />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Challenge 4', () => {
    const tree = renderer
      .create(<Challenge4 />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Challenge 5', () => {
    const tree = renderer
      .create(<Challenge5 />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Challenge 6', () => {
    const tree = renderer
      .create(<Challenge6 />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Challenge 7', () => {
    const tree = renderer
      .create(<Challenge7 />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Challenge 8', () => {
    const tree = renderer
      .create(<Challenge8 />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Challenge 9', () => {
    const tree = renderer
      .create(<Challenge9 />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Challenge 10', () => {
    const tree = renderer
      .create(<Challenge10 />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Challenge 11', () => {
    const tree = renderer
      .create(<Challenge11 />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Challenge 12', () => {
    const tree = renderer
      .create(<Challenge12 />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Challenge 13', () => {
    const tree = renderer
      .create(<Challenge13 />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
