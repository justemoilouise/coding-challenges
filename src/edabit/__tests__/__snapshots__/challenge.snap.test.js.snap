// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edabit challenges snapshots Challenge 1 1`] = `
<div>
  <div>
    Input: 
    <input />
    <button
      onClick={[Function]}
    >
      Convert
    </button>
  </div>
</div>
`;

exports[`Edabit challenges snapshots Challenge 2 1`] = `
<div
  className=""
>
  <div>
    Input: 
    <input
      min={0}
      type="number"
    />
    <button
      onClick={[Function]}
    >
      Submit
    </button>
  </div>
</div>
`;

exports[`Edabit challenges snapshots Challenge 3 1`] = `
<div
  className=""
>
  <div>
    <div>
      Input: 
      <input
        onKeyDown={[Function]}
        type="number"
      />
      <button
        onClick={[Function]}
      >
        Submit
      </button>
    </div>
  </div>
  <div>
    <div>
      Output:
    </div>
  </div>
</div>
`;

exports[`Edabit challenges snapshots Challenge 4 1`] = `
<div
  className=""
>
  <div>
    Input: 
    <input
      min={1}
      type="number"
    />
    <button
      onClick={[Function]}
    >
      Play
    </button>
  </div>
</div>
`;

exports[`Edabit challenges snapshots Challenge 5 1`] = `
<div
  className=""
>
  <div>
    <div>
      Input: 
      <input
        onKeyDown={[Function]}
        type="number"
      />
      <button
        onClick={[Function]}
      >
        Submit
      </button>
    </div>
  </div>
</div>
`;

exports[`Edabit challenges snapshots Challenge 6 1`] = `
<div
  className=""
>
  <div>
    Input: 
    <input
      min={1}
      type="number"
    />
    <button
      onClick={[Function]}
    >
      Submit
    </button>
  </div>
  <div>
    Output: 
    Yes
  </div>
</div>
`;
